### DISCUSSION TITLE
dynamic api routes

### DISCUSSION THREAD

**Discussion by philippdormann** [nuxt/nuxt#18323] [Ideas]
just [like in Next.js](https://nextjs.org/docs/api-routes/dynamic-api-routes) there should be dynamic api routes

e.g. `/api/users/[id].ts`

**Comment by wobsoriano** [comment]
https://github.com/unjs/h3/issues/4

**Reply by wobsoriano** [reply]
For now, I do it like this:

```ts
// server/api/users.ts
export default async (req: IncomingMessage) => {
    const { pathname } = new URL(req.url!, 'https://example.org/'); // just put it random url
    const id = pathname.split('/')[1];

    if (!id) {
        const result = await getAllUsers();
        return {
            result
        }
    }
 
    const result = await getUserById(id);
    return {
        result
    }
}
```

**Reply by philippdormann** [reply]
yeah, that's cool for now 👍

as soon as h3 supports it, the integrated path param support would be a great addition to nuxt framework tho 😀
(your code only works for basic api path params, not nested ones)

**Reply by wobsoriano** [reply]
yeah the example works for single level param only
